main.go

```diff
@@ -305,26 +305,40 @@ func readVcf(config *Config, reader *bufio.Reader, writer *bufio.Writer) {
 
 	var arrowWriter *bystroArrow.ArrowWriter
 	if config.dosageMatrixOutPath != "" {
-		sampleNames := header[sampleIdx:]
-		fieldNames := append([]string{"locus"}, sampleNames...)
+		if len(header) <= sampleIdx {
+			log.Print("No samples found in VCF file; writing empty dosage matrix file")
+			// Write empty file
+			file, err := os.Create(config.dosageMatrixOutPath)
```

Comment:
We log and create an empty file because we need to make sure downstream apps that expect the file don't crash, and to be consistent with what the caller of bystro-vcf (Bystro annotator typically) is asking for (bystro annotator specifies dosageOutput is wanted, and will report, when it returns, the path of that file).